experimenter/experimenter/experiments/models.py

```diff
@@ -1285,30 +1287,101 @@ def timeline(self):
 
         return timeline_entries
 
-    @property
-    def metric_areas(self):
-        metric_areas = NimbusConstants.DEFAULT_METRIC_AREAS.copy()
+    def metric_areas(self, analysis_basis, segment, reference_branch, window="overall"):
+        metric_areas = {}
+
+        notable_metrics = []
+        metric_areas[NimbusUIConstants.NOTABLE_METRIC_AREA] = notable_metrics
+        metric_areas[NimbusUIConstants.KPI_AREA] = self.get_kpi_metrics(
+            analysis_basis, segment, reference_branch, window
+        )
+
+        metrics_metadata = (
+            self.results_data.get("v3", {}).get("metadata", {}).get("metrics")
+            if self.results_data
+            else {}
+        ) or {}
+        all_outcome_metric_slugs = []
+        for slug in self.primary_outcomes + self.secondary_outcomes:
+            outcome = Outcomes.get_by_slug_and_application(slug, self.application)
+            metrics = outcome.metrics if outcome else []
+            outcomes = []
+
+            for metric in metrics:
+                formatted_metric = {
+                    "slug": metric.slug,
+                    "description": metric.description
+                    if metric.description
+                    else metrics_metadata.get(metric.slug, {}).get("description", ""),
+                    "group": "other_metrics",
+                    "friendly_name": metric.friendly_name
+                    if metric.friendly_name
+                    else metrics_metadata.get(metric.slug, {}).get(
+                        "friendly_name", metric.slug
```

Comment:
I think there is (was?) a util function that would convert something like `metric_slug` to `Metric Slug` that you could use here? Not a big deal though if you don't see it.